Makefile

```diff
@@ -15,6 +15,7 @@ ci:
 	echo "Running Go builds" && \
 	go build cmd/*.go && \
 	go build example/*.go && \
+	go test ./... -count=1 && \
```

Comment:
Nicer Makefile that lets you run the tests separately if you want:
```
pre-commit:
        echo "Running pre-commit" && \
        pre-commit run --all-files

build-all:
        echo "Running Go builds" && \
        go build cmd/*.go && \
        go build example/*.go

test:
        go test ./... -count=1

ci: pre-commit build-all test
        echo "All done"

.PHONY: lint format fmt ci pre-commit build-all test
```

Reply:
yeah i was thinking about this, it's also kinda nice to have tests run even if go fmt and go lint fail so i'll probably rework this a bit